R/conda.R

```diff
@@ -286,15 +286,28 @@ conda_install <- function(envname = NULL,
     }
   }
 
+  
   # delegate to pip if requested
   if (pip) {
+    conda_env_PATH_additions <- normalizePath(file.path(dirname(python), c(
```

Comment:
Should we remove paths that don't exist? (e.g. if things happened to change in a future version of Anaconda)

Is there any way we can ask Anaconda itself what these PATH entries are?

Reply:
Also, should we modify the PATH in the same way when initializing a copy of Python that lives in a Conda environment? We do some PATH munging here:

https://github.com/rstudio/reticulate/blob/277f4109094d36aa9865c66b8c870d567febd3ca/R/config.R#L487-L529

I suspect it would be best if we shared code in those two locations.

Reply:
> Should we remove paths that don't exist? (e.g. if things happened to change in a future version of Anaconda)

I don't think this is a good idea. I can imagine a scenario where conda might create one of these directories as part of whatever it's doing. Also, it's harmless to have non-existent directories on the PATH.

> Is there any way we can ask Anaconda itself what these PATH entries are?

The only reliable way I think would be to shell out with a `conda activate <envname>` command and report back the PATH, but even that get's tricky today because `conda activate` will fail with this message: 
```
CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.
To initialize your shell, run

    $ conda init <SHELL_NAME>

Currently supported shells are:
  - bash
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell

See 'conda init --help' for more information and options.

IMPORTANT: You may need to close and restart your shell after running 'conda init'.
```
The absence of `cmd.exe` from that list worries me; I think that's the default shell from R on windows.

Relevant: 
Default hook that get's run on 'conda init'
https://github.com/conda/conda/blob/master/conda/shell/etc/profile.d/conda.sh
Note there is a separate path on windows, maybe this PATH injection should be confined to windows only. 
https://github.com/conda/conda/blob/0b1312ce65bf0fbb8cbea3750f07c32a2492c57a/conda/shell/etc/profile.d/conda.sh#L14

(also, my heart goes out to whoever had to write this: https://github.com/conda/conda/blob/33a142c16530fcdada6c377486f1c1a385738a96/conda/shell/condabin/_conda_activate.bat)

I'll boot back into windows and experiment a little more with the suggested changes.

Have a nice weekend!